service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceQueries.kt

```diff
@@ -339,3 +357,63 @@ fun Database.Transaction.deleteCurrentDayAbsences(childId: UUID) {
         .bind("today", LocalDate.now(zoneId))
         .execute()
 }
+
+fun Database.Read.getChildSensitiveInfo(childId: UUID): ChildSensitiveInformation? {
+    val person = handle.getPersonById(childId)
+    val placementTypes = handle.getPlacementsForChild(childId).map { it.type }
+    val child = handle.getChild(childId)
+    val backupPickups = getBackupPickupsForChild(childId)
+    val familyContacts = fetchFamilyContacts(childId)
+
+    return if (person != null) {
+        ChildSensitiveInformation(
+            id = person.id,
+            firstName = person.firstName ?: "",
+            lastName = person.lastName ?: "",
+            preferredName = child?.additionalInformation?.preferredName ?: "",
+            ssn = person.identity.toString(),
+            childAddress = person.streetAddress ?: "",
+            placementTypes = placementTypes,
+            allergies = child?.additionalInformation?.allergies ?: "",
+            diet = child?.additionalInformation?.diet ?: "",
+            medication = child?.additionalInformation?.medication ?: "",
+            contact1 = familyContacts.getOrNull(0)?.let {
+                ContactInfo(
+                    firstName = it.firstName ?: "",
+                    lastName = it.lastName ?: "",
+                    phone = it.phone ?: "",
+                    backupPhone = it.backupPhone ?: "",
+                    email = it.email ?: ""
+                )
+            },
+            contact2 = familyContacts.getOrNull(1)?.let {
+                ContactInfo(
+                    firstName = it.firstName ?: "",
+                    lastName = it.lastName ?: "",
+                    phone = it.phone ?: "",
+                    backupPhone = it.backupPhone ?: "",
+                    email = it.email ?: ""
+                )
+            },
```

Comment:
The contacts that should be shown in mobile view are those that have a "priority" set, so it's not always a fixed 1-2 people, it could be more. The `fetchFamilyContacts(childId)` query also doesn't set default values.

Reply:
Revised after discussion